metric-example/metric-expressions-by-example.md

```diff
@@ -0,0 +1,248 @@
+to pair equal tuples for the calculation, or if you instead want all tuples of one side
+to be paired with a single, empty tuple on the other side (`:splitBy()`). Each combination
+has its own use cases.
+
+## Scenario 4: Response Time Categories
+> **Task**
+> Given a maximum acceptable response time of t = 1.5s in the 99th percentile, we want to
+> classify response times of our services as "satisfactory" (&le; t), "tolerable" (&le; 4t)
+> or "frustrating" (&gt; 4t). We want to see how the service count in each category changes
+> over time.
+
+_Note: This scenario uses the partition operator, which is available in Dynatrace versions 227 and later._
+
+First, we obtain a response time in the 99th percentile for each timeslot and every service,
+and we scale from microseconds to full seconds:
+```
+(builtin:service.response.time:percentile(99)) / (1000) / (1000)
+```
+
+Next up, we want to categorize each data point into one of the three categories. This is done
+by splitting up each series into up three new ones, and then assigning each data point to at
+most one of the three new series, depending on their value. When a data point has been assigned
+to one category, then the same timeslot will be `null` for the other categories. Let's use the
+`partition` operator to achieve just that:
+```
+: partition(
+    apdexCategory,
+    value("satisfactory", range(0, 1.5)),
+    value("tolerable",    range(1.5, 6)),
+    value("frustrating",  otherwise) 
+  )
+```
+
+Now, the reponse time series for each series belong to a new series with "dt.entity.service"
+and "apdexCategory" as their dimension. Now we want to merge all data with the same category. By specifying
+`:count` after the merge operation, instead of the values, we query just the data point count that
+made it into the merge of each timeslot:
+```
+: splitBy("apdexCategory")
+: count
+```
+
+That's almost what we want, but if not a single data point was available for a timeslot of a
+category, the value will be a data gap, which is represented with `null`. `0` would be more
+correct for our use case, so let's replace nulls with zero:
+```
+: default(0)
+```
+
+Putting it all together we get:
+```
+((builtin:service.response.time:percentile(99)) / (1000) / (1000))
+: partition(
+    apdexCategory,
+    value("satisfactory", range(0, 1.5)),
+    value("tolerable",    range(1.5, 6)),
+    value("frustrating",  otherwise) 
+  )
+: splitBy("apdexCategory")
+: count
+: default(0)
+```
+
+With the tools from this scenario at our disposal we can:
+* split up data points of series into categories,
+* count merged data points,
+* assign a default value for gaps in the data.
+
+## Scenario 5: Available Memory of Hosts with Top 10 PGIs
+> **Task** We recently increased the maximum heap space for all our Java processes to achieve
+> better utilization of available RAM. We want to check if the remaining memory of the hosts
```

Comment:
```suggestion
> better utilization of available RAM. We want to verifying that there's enough free memory on the hosts where the Top 10 most memory consuming processes run.
```

Reply:
Much better.

I made the change, but used "to verify" instead of "to verifying".